pkg/chunk/by_key.go

```diff
@@ -5,7 +5,36 @@ type ByKey []Chunk
 
 func (cs ByKey) Len() int           { return len(cs) }
 func (cs ByKey) Swap(i, j int)      { cs[i], cs[j] = cs[j], cs[i] }
-func (cs ByKey) Less(i, j int) bool { return cs[i].ExternalKey() < cs[j].ExternalKey() }
+func (cs ByKey) Less(i, j int) bool { return lessByKey(cs[i], cs[j]) }
+
+// This comparison uses all the same information as Chunk.ExternalKey()
+func lessByKey(a, b Chunk) bool {
+	if a.UserID < b.UserID {
+		return true
+	} else if a.UserID == b.UserID {
```

Comment:
Can we drop the else as the above clause returns?  Should bring the indentation in.